crates/swc_ecma_minifier/src/metadata/mod.rs

```diff
@@ -11,6 +14,37 @@ use swc_ecma_visit::{
 
 use crate::option::CompressOptions;
 
+#[derive(Debug, Eq)]
+struct HashEqIgnoreSpanExprRef<'a>(&'a Expr);
+
+impl<'a> PartialEq for HashEqIgnoreSpanExprRef<'a> {
+    fn eq(&self, other: &Self) -> bool {
+        Ident::within_ignored_ctxt(|| self.0.eq_ignore_span(other.0))
+    }
+}
+
+impl<'a> Hash for HashEqIgnoreSpanExprRef<'a> {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        // In pratice, most of cases/input we are dealing with are Expr::Member or
+        // Expr::Ident.
+        match self.0 {
+            Expr::Ident(i) => {
+                i.sym.hash(state);
+            }
+            Expr::Member(i) => {
+                Self(&i.obj).hash(state);
+                if let MemberProp::Ident(prop) = &i.prop {
+                    prop.sym.hash(state);
```

Comment:
I think there's a bug here that `console.log` will hash the same as the identifier `consolelog`. We'd need to hash `"."` or similar sigil.

Reply:
Does it matter? `Eq` will be used if the hash value matches

Reply:
Ohhhhhhh.